### PR TITLE
Repo cleanup + code cleanup + make text darkmode-compatible

### PR DESCRIPTION
**Pull request by twilight-flower**
Repo structure:
- Rename `main.py` to `glowfic-dl.py` for purposes of useful filename descriptiveness
- Add shebang to `glowfic-dl.py` to make it easier to call
- Add `Pipfile` for easy dependency-installation via `pipenv`, for use by those who prefer that over installing all the dependencies into their global environment
- Update readme to take into account all of the above

Code cleanup:
- Reorganize imports into alphabetical order, with Python standard library imports in a separate block from PyPI module imports
- Reorganize code from haphazard everything-in-arbitrary-order into clearly-labeled blocks
- Add type annotations to everything
- Refactor cookie-grabbing from `main()` into its own function
- Switch URL input from `sys.argv[1]` to an `argparse`-based function, to add a help menu and to improve error reporting on bad input
- Remove unused parameter `i` from `download_chapter`

Changes to generated files:
- Update stylesheet to avoid problems with white-on-white text in the Calibre reader when set to dark mode

Bug fixes:
- Add `tzdata` to dependencies list, since its absence was causing `zoneinfo` to throw errors on my machine
- Switch to a nondefault `asyncio` event loop policy conditional on the script being run on Windows, to eliminate a bug where the program would spam a dozen or so `RuntimeError: Event loop is closed`s errors into the terminal with tracebacks upon terminating successfully

### CODE REVIEW

File: glowfic-dl.py [hunk 1]
```diff
@@ -56,14 +56,14 @@
 }
 div.post {
```
Comment (rocurley) [inline glowfic-dl.py:57]: I want to make sure I understand the change here.

You've moved the grey borders between and around posts from being the background of the `posts` div to being the border of the `post` divs. To do this, you needed to move the spacing between posts from being a margin to being the border thickness. However, this means you lose margin collapse, so you manually get it back witbh the `div.post + div.post` rule.

Presumably, based on "make text darkmode-compatible", the goal here is to remove the `background: white` line, so the background gets set by the default css for dark mode.

Is my understanding correct?

**Review by rocurley — COMMENTED** [review]
This looks good, thanks.

Presumably a lot of the cleanup here was linter-guided. Could you include a way to invoke whatever tool you used to check your cleanup? Otherwise I expect this will rot.

File: glowfic-dl.py [hunk 2]
```diff
@@ -1,17 +1,23 @@
-from bs4 import BeautifulSoup
-from ebooklib import epub
-from tqdm.asyncio import tqdm
-import sys
-from urllib.parse import urljoin, urlparse
-from collections import OrderedDict
+#!/usr/bin/env python
```
Comment (rocurley) [inline glowfic-dl.py]: Hmm, this doesn't work for me without this change:
```suggestion
#!/usr/bin/env python3
```

File: glowfic-dl.py (hunk 1 above)
Comment (twilight-flower) [reply glowfic-dl.py:57]: The cleanup wasn't actually linter-guided; I just spent a couple hours manually cleaning things up before I started working on more-substantial changes, as a way of improving my own understanding of the code and how it flowed, and then went and ran `black` over it in order to keep it formatted in accordance with your patterns up to that point. So I can't point at any tools beyond the one I'm pretty sure you're already using, unfortunately.

Your understanding of the CSS change is correct!

Comment (rocurley) [reply glowfic-dl.py:57]: Oh ok, fair enough.

**Comment by rocurley** [comment]
Just tested on Kindle, Kobo and Calibre and the output is at least as good as it used to be, so I'm happy.

**Comment by rocurley** [comment]
Thanks!

**Pull request merged**